### DISCUSSION TITLE
[BE101] 寫留言板時，碰到 php 報錯 trying to access array offset on value of type null

### DISCUSSION THREAD

**Discussion by KW0409** [Lidemy/forum#139] [後端相關問題] [answered]
<img width="800" alt="截圖 2021-06-12 下午7 33 12" src="https://user-images.githubusercontent.com/80152099/121774639-ee273c80-cbb5-11eb-84fd-4f622de8c473.png">
<img width="800" alt="截圖 2021-06-12 下午7 33 12" src="https://user-images.githubusercontent.com/80152099/121774643-eff10000-cbb5-11eb-970c-deb9afba9d47.png">
<img width="800" alt="截圖 2021-06-12 下午6 49 55" src="https://user-images.githubusercontent.com/80152099/121774658-0303d000-cbb6-11eb-901b-f3f4ec23698a.png">

**Comment by KW0409** [comment, marked as answer]
可參考[連結1](https://blog.csdn.net/weixin_42193830/article/details/111460462)
可參考[連結2](http://jianyuluntan.com/post/158.html)

個人嘗試後發現用 !empty() 或 is_array() 先判斷值
都可以解決問題
(以下會用 !empty() 的方法)
原本的 `index.php`：
<img width="800" alt="截圖 2021-06-12 下午7 46 13" src="https://user-images.githubusercontent.com/80152099/146183083-b122c1a1-fc6b-45d0-b958-4ea7da9fc5f0.png">


修改後的 `index.php`：
<img width="800" alt="截圖 2021-06-12 下午7 46 13" src="https://user-images.githubusercontent.com/80152099/146183096-d3d7bafe-492a-484e-a0f0-64e46ae22710.png">

<br><br>


原本的 `utils.php`：
<img width="800" alt="截圖 2021-06-12 下午7 46 13" src="https://user-images.githubusercontent.com/80152099/146183115-a69bc223-0738-4ae9-abb4-99882381715b.png">


修改後的 `utils.php `：
<img width="800" alt="截圖 2021-06-12 下午7 46 13" src="https://user-images.githubusercontent.com/80152099/146183132-084efd32-1949-46d4-8e06-ceadd27662c2.png">

**Reply by outshaker** [reply]
你很棒，解完問題還特地把解法也發上來 :)

**Reply by KW0409** [reply]
哈哈沒有啦
我只是想說可能會有同學有碰到相關的問題，可以給對方提供一些參考

**Reply by ks0dcongra** [reply]
這個網站我有印象 我曾經也是在Lidemy˙學習的學生哈哈
我在維護我的個人網站時候也有碰到一樣的問題，謝謝你~
